src/helpers/global_session/global_session_context_provider.ts

```diff
@@ -0,0 +1,56 @@
+import { merge, map, type Subject, of, filter, tap, distinctUntilChanged } from 'rxjs';
+import { sha256 } from 'js-sha256';
+import moment from 'moment';
+import type { AnalyticsClient } from '../../client';
+
+export interface GlobalSessionContextProviderOpts {
+  /**
+   * The analytics client to register the context provider
+   */
+  analyticsClient: Pick<AnalyticsClient, 'registerContextProvider'>;
+  /**
+   * Observable of the current user id
+   */
+  userId$: Subject<string>;
+  /**
+   * Organization ID
+   */
+  organizationId?: string;
+}
+
+const SESSION_STORAGE_KEY = 'ebt_global_session';
+
+export function registerGlobalSessionContextProvider({
+  analyticsClient,
+  userId$,
+  organizationId,
+}: GlobalSessionContextProviderOpts) {
+  const sessionId$ = userId$.pipe(
+    filter((userId): userId is string => !!userId),
+    map((userId) => {
+      const startOfTheDay = moment().startOf('day').format('YYYY-MM-DD');
```

Comment:
This will act wonky with timezones. I recommend initializing moment date with utc. (note this is slightly different than `moment().utc()` in some cases too so be carefuld :D it bit me too many times. Maybe adding a dedicated unit test for timezones also makes sense since these bugs go unnoticed for a while


```suggestion
      const startOfTheDay = moment().utc().startOf('day').format('YYYY-MM-DD');
```